metricbeat/module/mongodb/status/status.go

```diff
@@ -47,24 +46,18 @@ func init() {
 // additional entries. These variables can be used to persist data or configuration between
 // multiple fetch calls.
 type MetricSet struct {
-	mb.BaseMetricSet
-	dialInfo *mgo.DialInfo
+	*mongodb.MetricSet
```

Comment:
This may be an invalid/stupid question because I'm still new to beats + golang but why do we need to embed the `*mongodb.MetricSet` type inside this `MetricSet` struct? Couldn't we get rid of the outer/wrapper struct now and in the `New` function below do this instead:

```
func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
    return mongodb.NewMetricSet(base)
}
```

Reply:
There are no stupid questions, this is a good one :)

Here all the metricsets share the same fields, so I use the same builder, but they need different implementations for the `Fetch()` method, so I embed the common data in structs implementing these specializations.